aten/src/ATen/native/Pool.h

```diff
@@ -205,8 +206,15 @@ pool3d_shape_check(
               "dilation should be greater than zero, but got ",
               "dilationT: ", dilationT, " dilationH: ", dilationH, " dilationW: ", dilationW);
 
-  TORCH_CHECK(input.numel() > 0 && (ndim == 4 || ndim == 5),
-              "non-empty 4D or 5D (batch mode) tensor expected for input, but got ndim: ", ndim);
+  TORCH_CHECK(ndim == 4 || ndim == 5,
+              fn_name, ": Expected 4D or 5D tensor for input with optional non-zero batch dim, but got ndim: ",
```

Comment:
nit: I think since this check is only concerned with the number of dims, "Expected 4D or 5D input tensor, but got ndim: " is better